synapse/tests/test_lib_snap.py

```diff
@@ -75,6 +77,87 @@ class SnapTest(s_t_utils.SynTest):
                 self.len(1, nodes)
                 self.eq(nodes[0], node)
 
+    async def test_addNodeRace(self):
+        ''' Test when a reader might retrieve a partially constructed node '''
+        NUM_TASKS = 2
+        failed = False
+        done_events = []
+        async with self.getTestCore() as core:
+
+            async def write_a_bunch(done_event):
+                nonlocal failed
+                data = list(range(50))
+                random.seed(4)  # chosen by fair dice roll
```

Comment:
We should be using an a instance of `random.Random()` instead of the random module functions directly.

Reply:
the instantiation would have to be out of scope of the `write_a_bunch` function to preserve the current test behavior

Reply:
the current code shuffles the data into the same ordered list for each task.